uvicorn/supervisors/watchgodreload.py

```diff
@@ -1,31 +1,43 @@
 import logging
+import os
 import re
 from pathlib import Path
+from socket import socket
+from typing import TYPE_CHECKING, Callable, List, Optional
 
 from watchgod import DefaultWatcher
 
+from uvicorn.config import Config
 from uvicorn.supervisors.basereload import BaseReload
 
 logger = logging.getLogger("uvicorn.error")
 
+if TYPE_CHECKING:
+    DirEntry = os.DirEntry[str]
+
 
 class CustomWatcher(DefaultWatcher):
     ignore_dotted_file_regex = r"^\/?(?:\w+\/)*(\.\w+)"
-    ignored = []
+    ignored: List[str] = []
 
-    def __init__(self, root_path):
+    def __init__(self, root_path: str) -> None:
         for t in self.ignored_file_regexes:
             self.ignored.append(t)
         self.ignored.append(self.ignore_dotted_file_regex)
         self._ignored = tuple(re.compile(r) for r in self.ignored)
         super().__init__(root_path)
 
-    def should_watch_file(self, entry):
+    def should_watch_file(self, entry: "DirEntry") -> bool:
```

Comment:
seems weird to me, but it's 1st time I see DirEntry, I would have thought `should_watch_file` entry parameter would be a file, the type name suggests it  is a directory,  DirEntry can be a file ?

Reply:
Yeah, I'm just honoring the parent class here.

Reference: https://github.com/samuelcolvin/watchgod/blob/1f738352da0a38eed25f4730fde097f06eaaacd6/watchgod/watcher.py#L90-L98

Reply:
But yes, it can.

Reply:
ok that's cool then